cns/middlewares/k8sSwiftV2.go

```diff
@@ -135,150 +134,62 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
+		return nil, errors.Wrapf(err, "failed to get pod's mtpnc from cache")
 	}
 
 	// Check if the MTPNC CRD is ready. If one of the fields is empty, return error
-	if mtpnc.Status.PrimaryIP == "" || mtpnc.Status.MacAddress == "" || mtpnc.Status.NCID == "" || mtpnc.Status.GatewayIP == "" {
-		return cns.PodIpInfo{}, errMTPNCNotReady
+	if !mtpnc.IsReady() {
+		return nil, errMTPNCNotReady
 	}
 	logger.Printf("[SWIFTv2Middleware] mtpnc for pod %s is : %+v", podInfo.Name(), mtpnc)
 
 	// Parse MTPNC primaryIP to get the IP address and prefix length
 	p, err := netip.ParsePrefix(mtpnc.Status.PrimaryIP)
 	if err != nil {
-		return cns.PodIpInfo{}, errors.Wrapf(err, "failed to parse mtpnc primaryIP %s", mtpnc.Status.PrimaryIP)
+		return nil, errors.Wrapf(err, "failed to parse mtpnc primaryIP %s", mtpnc.Status.PrimaryIP)
 	}
 	// Get the IP address and prefix length
 	ip := p.Addr()
 	prefixSize := p.Bits()
 	if prefixSize != prefixLength {
-		return cns.PodIpInfo{}, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
+		return nil, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
 	}
-	podIPInfo := cns.PodIpInfo{
-		PodIPConfig: cns.IPSubnet{
-			IPAddress:    ip.String(),
-			PrefixLength: uint8(prefixSize),
-		},
-		MacAddress:        mtpnc.Status.MacAddress,
-		NICType:           cns.DelegatedVMNIC,
-		SkipDefaultRoutes: false,
-		// InterfaceName is empty for DelegatedVMNIC
-	}
-
-	return podIPInfo, nil
-}
 
-// setRoutes sets the routes for podIPInfo used in SWIFT V2 scenario.
-func (m *K8sSWIFTv2Middleware) setRoutes(podIPInfo *cns.PodIpInfo) error {
-	logger.Printf("[SWIFTv2Middleware] set routes for pod with nic type : %s", podIPInfo.NICType)
-	podIPInfo.Routes = []cns.Route{}
-	switch podIPInfo.NICType {
-	case cns.DelegatedVMNIC:
-		virtualGWRoute := cns.Route{
-			IPAddress: fmt.Sprintf("%s/%d", virtualGW, prefixLength),
-		}
-		// default route via SWIFT v2 interface
-		route := cns.Route{
-			IPAddress:        "0.0.0.0/0",
-			GatewayIPAddress: virtualGW,
-		}
-		podIPInfo.Routes = []cns.Route{virtualGWRoute, route}
-	case cns.InfraNIC:
-		// Get and parse infraVNETCIDRs from env
-		infraVNETCIDRs, err := configuration.InfraVNETCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
-		}
-		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
-		}
+	podIPInfos := make([]cns.PodIpInfo, len(mtpnc.Status.InterfaceInfos))
 
-		// Get and parse podCIDRs from env
-		podCIDRs, err := configuration.PodCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get podCIDRs from env")
-		}
-		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podCIDRs")
+	// Set the first element from the first InterfaceInfos
```

Comment:
why we set first element from the first interfaceInfos and then fill rest of them. Cannot we fill all elements from interfaceInfos to podIPInfos once?

Reply:
no Matt said we have to keep this same behavior for DNC-RC, we will not deprecate old fields but populate in both places instead

Reply:
this changes now.. we will either read the array or create single podipinfo